test/data_base.cpp

```diff
@@ -1,401 +1,1148 @@
+   settings.index_addresses = false;
+   settings.flush_writes = false;
+   settings.file_growth_rate = 42;
+   settings.block_table_buckets = 42;
+   settings.transaction_table_buckets = 42;
+   settings.address_table_buckets = 42;
+
+   data_base_accessor instance(settings);
+
+   static const auto bc_settings = bc::settings(bc::config::settings::mainnet);
+   const chain::block genesis = bc_settings.genesis_block;
+   BOOST_REQUIRE(instance.create(genesis));
+
+   const auto block1 = read_block(MAINNET_BLOCK1);
+   store_block_transactions(instance, block1, 1);
+
+   BOOST_REQUIRE_EQUAL(instance.push_header(block1.header(), 1, 100), error::success);
+   BOOST_REQUIRE_EQUAL(instance.candidate(block1), error::success);
+   BOOST_REQUIRE_EQUAL(instance.push_block(block1, 1), error::success);
+
+   // setup ends
+
+   chain::block out_block;
+   BOOST_REQUIRE_EQUAL(instance.pop_block(out_block, 1), error::success);
+
+   // test conditions
+
+   BOOST_REQUIRE(out_block.hash() == block1.hash());
+   test_block_not_exists(instance, block1, settings.index_addresses);
+   test_heights(instance, 1u, 0u);
+}
+
+BOOST_AUTO_TEST_CASE(data_base__push_all_and_update__already_candidated___success)
+{
+   create_directory(DIRECTORY);
+   database::settings settings;
+   settings.directory = DIRECTORY;
+   settings.index_addresses = false;
+   settings.flush_writes = false;
+   settings.file_growth_rate = 42;
+   settings.block_table_buckets = 42;
+   settings.transaction_table_buckets = 42;
+   settings.address_table_buckets = 42;
+  
+   data_base_accessor instance(settings); 
+   
+   static const auto bc_settings = bc::settings(bc::config::settings::mainnet);
+   const chain::block genesis = bc_settings.genesis_block;
+   BOOST_REQUIRE(instance.create(genesis));
+   
+   const auto block1 = read_block(MAINNET_BLOCK1);
+
+   block_const_ptr block1_ptr = std::make_shared<const message::block>(read_block(MAINNET_BLOCK1));
+   block_const_ptr block2_ptr = std::make_shared<const message::block>(read_block(MAINNET_BLOCK2));
+   block_const_ptr block3_ptr = std::make_shared<const message::block>(read_block(MAINNET_BLOCK3));
+   const auto blocks_push_ptr = std::make_shared<const block_const_ptr_list>(block_const_ptr_list{
+           block1_ptr, block2_ptr, block3_ptr });
+   store_block_transactions(instance, *block1_ptr, 1);
+   store_block_transactions(instance, *block2_ptr, 1);
+   store_block_transactions(instance, *block3_ptr, 1);
+
+   const auto headers_push_ptr = std::make_shared<const header_const_ptr_list>(header_const_ptr_list{
+           std::make_shared<const message::header>(block1_ptr->header()),
+               std::make_shared<const message::header>(block2_ptr->header()),
+               std::make_shared<const message::header>(block3_ptr->header())
+               });
+      
+   BOOST_REQUIRE(instance.push_all(headers_push_ptr, config::checkpoint(genesis.hash(), 0)));
+   for(const auto block_ptr: *blocks_push_ptr) {
```

Comment:
Style: whitespace (“for” is not a fn), balance braces.

Reply:
Fixed everywhere.